### PR TITLE
feat(database-change): add and adapt the current schedule interface

### PR DESCRIPTION
**Pull request by kiko-art**
#### What type of PR is this?
type-feature

#### What this PR does / why we need it:
1. The existing interface accommodates the database change.
2. Two new interfaces were added to delete schedule and list all tasks
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/ScheduleTaskOverview.java [hunk 1]
```diff
@@ -41,6 +49,10 @@ public class ScheduleTaskOverview {
 
     private Date updateTime;
 
+    private Date lastFireTime;
```
Comment (guowl3) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/ScheduleTaskOverview.java:52]: lastExecutionTime

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/ScheduleTaskOverview.java [hunk 2]
```diff
@@ -33,6 +35,12 @@ public class ScheduleTaskOverview {
 
     private Long id;
 
+    private String scheduleId;
```
Comment (guowl3) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/ScheduleTaskOverview.java:38]: Use a new object to represent the task list from the execution perspective

File: server/odc-service/src/main/java/com/oceanbase/odc/service/sqlplan/model/SqlPlanParameters.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import com.oceanbase.odc.service.schedule.model.ScheduleTaskParameters;
+
+import lombok.Data;
+
+@Data
+public class SqlPlanParameters implements ScheduleTaskParameters {
+
+    private Long databaseId;
+
+    private Database databaseInfo;// 包含数据源、数据库信息
```
Comment (guowl3) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/sqlplan/model/SqlPlanParameters.java:31]: @JsonProperty(access = Access.READ_ONLY)

File: server/odc-service/src/main/java/com/oceanbase/odc/service/sqlplan/model/SqlPlanParameters.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+    private boolean modifyTimeoutIfTimeConsumingSqlExists = true;
+
+    private Integer retryTimes = 0;
+
+    private Long retryIntervalMillis = 180000L;
+
+    private TaskErrorStrategy errorStrategy;
+
+    private Integer riskLevelIndex;
```
Comment (guowl3) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/sqlplan/model/SqlPlanParameters.java:53]: What is the purpose of the 'riskLevelIndex' field?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/sqlplan/model/SqlPlanParameters.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    private Long timeoutMillis = 172800000L;
+
+    private String delimiter = ";";
+
+    private Integer queryLimit = 1000;
+
+    private boolean modifyTimeoutIfTimeConsumingSqlExists = true;
+
+    private Integer retryTimes = 0;
```
Comment (guowl3) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/sqlplan/model/SqlPlanParameters.java:47]: please add a description for the field, such as 'number of retry attempts when SQL execution fails'

File: server/odc-service/src/main/java/com/oceanbase/odc/metadb/schedule/ScheduleRepository.java [hunk 6]
```diff
@@ -68,7 +68,21 @@ default Page<ScheduleEntity> find(@NotNull Pageable pageable, @NotNull QuerySche
...
+
+    default List<ScheduleEntity> find(@NotNull QueryScheduleParams params) {
+        Specification<ScheduleEntity> specification = Specification
+                .where(OdcJpaRepository.between(ScheduleEntity_.createTime, params.getStartTime(), params.getEndTime()))
+                .and(OdcJpaRepository.in(ScheduleEntity_.dataSourceId, params.getDataSourceIds()))
+                .and(OdcJpaRepository.eq(ScheduleEntity_.databaseName, params.getDatabaseName()))
+                .and(OdcJpaRepository.eq(ScheduleEntity_.type, params.getType()))
+                .and(OdcJpaRepository.in(ScheduleEntity_.projectId, params.getProjectIds()))
+                .and(OdcJpaRepository.eq(ScheduleEntity_.id, params.getId()))
+                .and(OdcJpaRepository.notEq(ScheduleEntity_.status, ScheduleStatus.DELETED))
```
[the remark this answers is not in the document]
Comment (kiko-art) [reply server/odc-service/src/main/java/com/oceanbase/odc/metadb/schedule/ScheduleRepository.java:84]: In the interface of deleting a job, the state of DELETED was added to indicate logical deletion. The deleted data should not be displayed in the list.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/sqlplan/model/SqlPlanParameters.java (hunk 5 above)
Comment (kiko-art) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/sqlplan/model/SqlPlanParameters.java:47]: Fields that are not related to the interface are eliminated.

**Review by guowl3 — APPROVED** [review]
lgtm

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduleService.java [hunk 7]
```diff
@@ -616,6 +633,53 @@ public Page<ScheduleTaskOverview> listScheduleTaskOverview(@NotNull Pageable pag
         return scheduleTaskService.getScheduleTaskListResp(pageable, scheduleId);
     }
 
+    public Page<ScheduleTaskListOverview> listScheduleTaskOverviewByScheduleType(@NotNull Pageable pageable,
+            @NotNull QueryScheduleTaskParams params) {
+        log.info("List schedule task overview req:{}", params);
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduleService.java:638]: use key=value for log message style.

`params={}`

File: server/odc-service/src/main/java/com/oceanbase/odc/service/sqlplan/model/SqlPlanTaskResult.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.sqlplan.model;
+
+import lombok.Data;
+
+@Data
+public class SqlPlanTaskResult {
+
+    private Long totalFireCount;
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/sqlplan/model/SqlPlanTaskResult.java:23]: what is fire count? is it task count or sql statement count?

if task, class may naming to SqlPlanScheduleResult.

if sql statement count, may:

totalStatements, finishedStatements, ...

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_3__alter_schedule_schedule.sql [hunk 9]
```diff
@@ -0,0 +1,4 @@
+--
+-- Add column `name` to `schedule_schedule` table
+--
+alter table `schedule_schedule` add column `name` varchar(255) DEFAULT NULL COMMENT '任务名称';
```
[the remark this answers is not in the document]
Comment (kiko-art) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_3__alter_schedule_schedule.sql:4]: column comment 'schedule name'

**Review by yizhouxw — APPROVED** [review]
lgtm

**Review by guowl3 — APPROVED** [review]

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**